func.go

```diff
@@ -133,6 +133,11 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				} else {
 					stack++
 				}
+			case reflect.Struct:
+				if runtime.GOOS != "darwin" || (runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64") {
+					panic("purego: struct arguments are only supported on macOS amd64 & arm64")
+				}
+				// TODO: implement
```

Comment:
@hajimehoshi What are your thoughts about having functions with struct arguments panicking only when they are called? There is already ~250 LOC solely for placing struct arguments. This supports _most_ cases but not even all so more code would be necessary to support them. (Although IDK if that needs to be done in this PR). In order to do the same "does it fit" when creating a function like other arguments all the struct code would basically have to be duplicated and it is tricky code. I'm suggesting we loosen the check for structs to only panic when the C function is about to be called.

Reply:
Did you mean that it is pretty troublesome to check whether a struct is available for purego or not? Can we have a function to check this by refactoring the below code...?

Reply:
That might work. I'll think about it